src/Impl/GuzzleHttpRequester.php

```diff
@@ -44,27 +59,37 @@ public function executePost(
      * @return MessageStatus[]
      * @throws StreamxClientException
      */
-    private function handleResponse(ResponseInterface $response): array
+    private function handleIngestionResponse(ResponseInterface $response): array
     {
         $statusCode = $response->getStatusCode();
 
-        if ($statusCode == 202) {
-            return $this->parseMessageStatuses($response);
-        }
-
-        if ($statusCode == 401) {
-            throw new StreamxClientException('Authentication failed. Make sure that the given token is valid.');
+        switch ($statusCode) {
```

Comment:
The same, but in switch format, since it fitted well also in the new `handleSchemaResponse` method